### PR TITLE
[zh-cn]: add translation of promise.try

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

As the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

None.

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
+---
+title: Promise.try()
+slug: Web/JavaScript/Reference/Global_Objects/Promise/try
+l10:
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:4]:
```suggestion
l10n:
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+Promise.try(func)
+Promise.try(func, arg1)
+Promise.try(func, arg1, arg2)
+Promise.try(func, arg1, arg2, /* …, */ argN)
+```
+
+### 参数
+
+- `func`
+  - ：使用提供的参数（`arg1`，`arg2`，...，`argN`）同步调用的函数。它可以做任何事情——要么返回一个值、要么抛出一个错误或者返回一个承诺。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:24]:
```suggestion
  - ：使用提供的参数（`arg1`，`arg2`，...，`argN`）同步调用的函数。它可以做任何事情——要么返回一个值、要么抛出一个错误或者返回一个 promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+{{Specifications}}
+
+## 浏览器兼容性
+
+{{Compat}}
+
+## 参见
+
+- [`core-js` 中 `Promise.try` 的 polyfill](https://github.com/zloirock/core-js#promisetry)
+- [使用 promises](/zh-CN/docs/Web/JavaScript/Guide/Using_promises) guide
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:174]:
```suggestion
- [使用 promise](/zh-CN/docs/Web/JavaScript/Guide/Using_promises) 指南
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+Promise.try(func, arg1)
+Promise.try(func, arg1, arg2)
+Promise.try(func, arg1, arg2, /* …, */ argN)
+```
+
+### 参数
+
+- `func`
+  - ：使用提供的参数（`arg1`，`arg2`，...，`argN`）同步调用的函数。它可以做任何事情——要么返回一个值、要么抛出一个错误或者返回一个承诺。
+- `arg1`, `arg2`, …, `argN`
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:25]:
```suggestion
- `arg1`、`arg2`、…、`argN`
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+一个 {{jsxref("Promise")}}，其状态可以是：
+
+- 已兑现的，如果 `func` 同步地返回一个值。
+- 已拒绝的，如果 `func` 同步地抛出一个错误。
+- 异步兑现或拒绝的，如果 `func` 返回一个 promise。
+
+## 描述
+
+你可能有一个接受一个回调函数的 API，这个回调函数可能是同步的，也可能是异步的。你希望将结果封装成一个 promise，统一处理。最直接的方法可能是 {{jsxref("Promise/resolve", "Promise.resolve(func())")}}。问题是，如果 `func()` 同步地抛出一个错误，这个错误不会被捕获，并变成一个被拒绝的 promise。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:38]:
```suggestion
你可能有一个接受回调函数的 API，这个回调函数可能是同步的，也可能是异步的。你希望将结果封装成一个 promise，统一处理。最直接的方法可能是 {{jsxref("Promise/resolve", "Promise.resolve(func())")}}。问题是，如果 `func()` 同步地抛出一个错误，这个错误不会被捕获，并变成一个被拒绝的 promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+new Promise((resolve) => resolve(func()));
+```
+
+但在这里，`Promise.try()` 更有用：
+
+```js
+Promise.try(func);
+```
+
+对于内置的 `Promise()` 构造函数，如果执行器函数抛出错误，它会自动捕获并将其转换为拒绝的 只是 `Promise.try()` 更简洁易读。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:52]:
```suggestion
对于内置的 `Promise()` 构造函数，如果执行器函数抛出错误，它会自动捕获并将其转换为拒绝状态，因此这两种方法基本等同，只是 `Promise.try()` 更简洁易读。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+请注意，`Promise.try()` 与下面代码并*不*完全等价，尽管它们非常相似：
+
+```js
+Promise.resolve().then(func);
+```
+
+它们的区别在于，传入给 `then()` 的回调总是异步调用的，而 `Promise()` 构造函数的执行器函数总是同步调用的。`Promise.try()` 也会同步调用函数，并立即解析 promise，如果可以的话。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:60]:
```suggestion
它们的区别在于，传入给 {{jsxref("Promise/then", "then()")}} 的回调总是异步调用的，而 `Promise()` 构造函数的执行器函数总是同步调用的。`Promise.try()` 也会同步调用函数，并尽可能立即解决 promise。
```

File: files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+`Promise.try()` 与 {{jsxref("Promise/catch", "catch()")}} 和 {{jsxref("Promise/finally", "finally()")}} 结合使用，可以用来在单个链中处理同步和异步错误，并使处理 promise 错误看起来像处理同步错误。
+
+类似 {{domxref("Window/setTimeout", "setTimeout()")}}， `Promise.try()` 接受额外的参数，这些参数将传递给回调。这意味着，不要这样做：
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/global_objects/promise/try/index.md:64]:
```suggestion
类似 {{domxref("Window/setTimeout", "setTimeout()")}}，`Promise.try()` 接受额外的参数，这些参数将传递给回调。这意味着，不要这样做：
```

**Review by jasonren0403 — COMMENTED** [review]
一些建议

**Review by jasonren0403 — APPROVED** [review]
👍

**Pull request merged**